shared/actions/config/index.tsx

```diff
@@ -280,6 +281,14 @@ const switchRouteDef = (
           ? [RouteTreeGen.createNavigateAppend({path: ['signupEnterPhoneNumber']})]
           : []),
       ]
+    } else if (action.type === ConfigGen.loggedIn) {
+      if (
+        state.config.startupLink &&
+        state.config.startupLink.endsWith('/phone-app') &&
+        !state.settings.phoneNumbers.phones
+      ) {
+        return [SettingsGen.createLoadSettings()]
```

Comment:
this should only happen once and with the logic that already does this kind of stuff
https://github.com/keybase/client/blob/master/shared/actions/config/index.tsx#L399